Удаление рекомендации

s3nkyh opened the pull request.

Commit:
added: RecomContrl, RecomDto, RecomService, SkillOfferDto

Commit:
--

Commit:
-

Commit:
--

Commit:
исправил RecommendationMapper

Commit:
убрал лишние аннотации

Commit:
исправил RecommendationMapper
добавил метод deleteAllAndCreate в RecommendationService

Commit:
реализовал механику удаления рекомендации

antonykapitukov commented on src/main/java/school/faang/user_service/controller/RecommendationController.java:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    public void deleteRecommendation(Long id) {
+        recommendationService.delete(id);
```
рекомендую добавить логи Slf4j на выполнение операций например при выполнении операции удаления выводиться сообщение об успешном удалении + его id

antonykapitukov commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    public void delete(Long id) {
+        recommendationRepository.deleteById(id);
```
рекомендую проверить id на null или написать аннотацию @NonNull из ломбока

Commit:
исправил маппер и валидацию

Commit:
добавил логи и проверку на null

Commit:
исправил RecommendationService,
RecommendationValidation,
RecommendationMapper
добавил метод в RecommendationRepository и SkillOfferRepository
